### ISSUE TITLE
Transaction gets stuck in tx pool

### ISSUE THREAD

**Issue by athei** [paritytech/contract-issues#19]
We are not able to reproduce it this, yet. But sometimes it happened that a transaction gets stuck in the transaction pool. It won't get included but also not removed from the pool. This will block the wallet from submitting any further transactions because the node will reject a new transaction with the same nonce but same tip.

It is unclear what is going on. It happens rarely. We need to reproduce the issue.

**Comment by michalkucharczyk** [comment]
This one is known for the old pool.

Old pool badly manages the transaction in case of many forks. Valid transactions can be rejected because on the other forks they can be staled or future. Here are some links [1](https://github.com/paritytech/polkadot-sdk/issues/1202#issuecomment-2070847339),[2](https://hackmd.io/kT8mquVcQfqn9F48U_w2rA#Problems).

On top of that. the transaction that is rejected on the collator/validator, will be stuck on RPC node, becuase RPC node never executes it. I think you observe this.

The fork-aware pool should solve the problem in most cases (by properly handling valid transactions on numerous forks).

**Comment by pgherveou** [comment]
@michalkucharczyk  is this "fork-aware pool" scheduled to go live soon? 
Could we have it deployed on westend?

**Comment by skunert** [comment]
>  is this "fork-aware pool" scheduled to go live soon?Could we have it deployed on westend? 

The pool is already available on nodes via `--pool-type fork-aware`, so for sure we could enable this for westend.

> dumb-question: are transactions getting stuck on relay-chain or is it parachain?

I assume this is about asset-hub-westend

**Comment by athei** [comment]
> > dumb-question: are transactions getting stuck on relay-chain or is it parachain?
> 
> I assume this is about asset-hub-westend

Correct its stuck on a parachain.

**Comment by skunert** [comment]
Had a quick chat with @michalkucharczyk and we agreed to enable it on ah-westend.

**Comment by michalkucharczyk** [comment]
@athei @pgherveou: do you want to coordinate this somehow?
Or should we just enable new pool and restart nodes at any time?

**Comment by athei** [comment]
Not really. But just wait until the morning in case something goes wrong. Thanks.

**Comment by pgherveou** [comment]
Yes anytime is good !

**Comment by michalkucharczyk** [comment]
sorry, not yet. will open issue now.

**Comment by michalkucharczyk** [comment]
https://github.com/paritytech/devops/issues/3760

To cross-check that new pool was enabled, you can take a look on the logs:
```
 [Parachain] maintain: txs:(0, 0) views:[1;[(291, 0, 0)]] event:NewBestBlock { hash: 0x6049755da98e21f4ebbdb5d31c59cf62b3b143d3a2aea69bd46ec0eab8fda9b7, tree_route: None }  took:461.031µs
```
or look into [metrics](https://github.com/paritytech/polkadot-sdk/blob/f845a9f42614120c98582f598d45d6d831455305/substrate/client/transaction-pool/src/fork_aware_txpool/metrics.rs#L31) for new pool specific keys.

I think it is not possible to check it from the RPC API perspective.

**Comment by michalkucharczyk** [comment]
The nodes were updated. Let me know if you see any issues with transactions processing.

**Comment by michalkucharczyk** [comment]
> or look into [metrics](https://github.com/paritytech/polkadot-sdk/blob/f845a9f42614120c98582f598d45d6d831455305/substrate/client/transaction-pool/src/fork_aware_txpool/metrics.rs#L31) for new pool specific keys

There is also grafana dashboard:
https://grafana.teleport.parity.io/goto/nEbz4TOHg?orgId=1

**Comment by pgherveou** [comment]
Let's close the issue for now and re-open it if we notice anything again
